Explorer++/Helper/Bookmark.cpp

```diff
@@ -295,40 +335,43 @@ void CBookmarkFolder::SerializeToRegistry(const std::wstring &strKey)
 	LONG lRes = RegCreateKeyEx(HKEY_CURRENT_USER,strKey.c_str(),
 	0,NULL,REG_OPTION_NON_VOLATILE,KEY_WRITE,NULL,&hKey,NULL);
 
-	if(lRes == ERROR_SUCCESS)
-	{
-		TCHAR guidString[128];
-		StringFromGUID2(m_guid,guidString,SIZEOF_ARRAY(guidString));
-		NRegistrySettings::SaveStringToRegistry(hKey,_T("GUID"),guidString);
-		NRegistrySettings::SaveStringToRegistry(hKey,_T("Name"),m_strName.c_str());
-		NRegistrySettings::SaveDwordToRegistry(hKey,_T("DateCreatedLow"),m_ftCreated.dwLowDateTime);
-		NRegistrySettings::SaveDwordToRegistry(hKey,_T("DateCreatedHigh"),m_ftCreated.dwHighDateTime);
-		NRegistrySettings::SaveDwordToRegistry(hKey,_T("DateModifiedLow"),m_ftModified.dwLowDateTime);
-		NRegistrySettings::SaveDwordToRegistry(hKey,_T("DateModifiedHigh"),m_ftModified.dwHighDateTime);
-
-		int iItem = 0;
-
-		for each(auto Variant in m_ChildList)
-		{
-			TCHAR szSubKey[256];
+	if (lRes != ERROR_SUCCESS)
+		return;
+
+	TCHAR guidString[128];
+	StringFromGUID2(m_guid, guidString, SIZEOF_ARRAY(guidString));
+	NRegistrySettings::SaveStringToRegistry(hKey, _T("GUID"), guidString);
+	NRegistrySettings::SaveStringToRegistry(hKey, _T("Name"), m_strName.c_str());
+	NRegistrySettings::SaveDwordToRegistry(hKey, _T("DateCreatedLow"), m_ftCreated.dwLowDateTime);
+	NRegistrySettings::SaveDwordToRegistry(hKey, _T("DateCreatedHigh"), m_ftCreated.dwHighDateTime);
+	NRegistrySettings::SaveDwordToRegistry(hKey, _T("DateModifiedLow"), m_ftModified.dwLowDateTime);
+	NRegistrySettings::SaveDwordToRegistry(hKey, _T("DateModifiedHigh"), m_ftModified.dwHighDateTime);
+
+	int iItem = 0;
 
-			if(CBookmarkFolder *pBookmarkFolder = boost::get<CBookmarkFolder>(&Variant))
-			{
-				StringCchPrintf(szSubKey,SIZEOF_ARRAY(szSubKey),_T("%s\\BookmarkFolder_%d"),strKey.c_str(),iItem);
-				pBookmarkFolder->SerializeToRegistry(szSubKey);
-			}
-			else if(CBookmark *pBookmark = boost::get<CBookmark>(&Variant))
-			{
-				StringCchPrintf(szSubKey,SIZEOF_ARRAY(szSubKey),_T("%s\\Bookmark_%d"),strKey.c_str(),iItem);
+	for each(auto Variant in m_ChildList)
+	{
+		TCHAR szSubKey[256];
 
-				/* TODO: Serialize. */
-			}
+		if (CBookmarkFolder *pBookmarkFolder = boost::get<CBookmarkFolder>(&Variant))
+		{
+			StringCchPrintf(szSubKey, SIZEOF_ARRAY(szSubKey), _T("%s\\BookmarkFolder_%d"), strKey.c_str(), iItem);
+			pBookmarkFolder->SerializeToRegistry(szSubKey);
+		}
+		else if (CBookmark *pBookmark = boost::get<CBookmark>(&Variant))
+		{
+			StringCchPrintf(szSubKey, SIZEOF_ARRAY(szSubKey), _T("Bookmark_%d"), iItem);
+			
+			NBookmark::SerializedData_t data = pBookmark->Serialize();
+			NRegistrySettings::SaveBinToRegistry(hKey, szSubKey, reinterpret_cast<LPBYTE>(data.pData), data.uSize);

```

Comment:
Rather than saving the serialized data to the registry, I think it would be better if the individual fields were saved (as is done for other data saved to the registry). That would also help to prevent issues where the binary data can't be used because the bookmark structure has changed.

The binary bookmark serialization was mainly added to allow bookmark data to be sent between Explorer++ processes.